### ISSUE TITLE
DoubleNegation cop and only_if { !!string }  use case

### ISSUE THREAD

**Issue by rmoriz** [chef/cookstyle#28]
see https://github.com/chef/chef/issues/5581

TL;DR

```ruby
resource "dummy" do
   not_if {  node['setting'] } 
end
```
throws an chef deprecation warning, however some use cases just exactly want to do that.

In https://github.com/chef/chef/issues/5581 there was the suggestion to use…

```ruby
resource "dummy" do
   not_if { !!node['setting'] } 
end
```

… however this currently triggers the `Style/DoubleNegation` cop.

```
Chef Development Kit Version: 1.2.22
chef-client version: 12.19.36
delivery version: master (0b746cafed65a9ea1a79de3cc546e7922de9187c)
berks version: 5.6.4
kitchen version: 1.16.0
chef -v  8,91s user 1,68s system 93% cpu 11,314 total

…

chef exec cookstyle acme_persistence.rb  -D
Inspecting 1 file
C

Offenses:

acme_persistence.rb:118:17: C: Style/DoubleNegation: Avoid the use of double negation (!!).
      only_if { !!item['crt'] }

…

**Comment by lamont-granquist** [comment]
chefstyle has it turned off:

https://github.com/chef/chefstyle/blob/master/config/chefstyle.yml#L498-L500

**Comment by rmoriz** [comment]
added PR if it helps…
